### ISSUE TITLE
Uninitialized constant traject in production

### ISSUE THREAD

**Issue by banukutlu** [psu-libraries/psulib_blacklight#95]
In GitLab by @cdm32 on Sep 6, 2018, 14:39

related to https://groups.google.com/forum/#!topic/blacklight-development/HrpLxDnNxww

```log
App 16305 output: Error: The application encountered the following error: uninitialized constant Traject (NameError)
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/bootsnap-1.3.1/lib/bootsnap/load_path_cache/core_ext/active_support.rb:53:in `block in load_missing_constant'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/bootsnap-1.3.1/lib/bootsnap/load_path_cache/core_ext/active_support.rb:8:in `without_bootsnap_cache'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/bootsnap-1.3.1/lib/bootsnap/load_path_cache/core_ext/active_support.rb:53:in `rescue in load_missing_constant'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/bootsnap-1.3.1/lib/bootsnap/load_path_cache/core_ext/active_support.rb:42:in `load_missing_constant'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/traject-2.3.4/lib/traject/null_writer.rb:3:in `<main>'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/bootsnap-1.3.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/bootsnap-1.3.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `block in require_with_bootsnap_lfi'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/bootsnap-1.3.1/lib/bootsnap/load_path_cache/loaded_features_index.rb:65:in `register'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/bootsnap-1.3.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:20:in `require_with_bootsnap_lfi'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/bootsnap-1.3.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/bundler/gems/blacklight-marc-d0176744e70f/lib/blacklight/marc/indexer.rb:2:in `<main>'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/bootsnap-1.3.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/bootsnap-1.3.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `block in require_with_bootsnap_lfi'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/bootsnap-1.3.1/lib/bootsnap/load_path_cache/loaded_features_index.rb:65:in `register'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/bootsnap-1.3.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:20:in `require_with_bootsnap_lfi'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/bootsnap-1.3.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'
App 16305 output:     /data/websites/psulib_blacklight/app/models/marc_indexer.rb:2:in `<main>'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/bootsnap-1.3.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/bootsnap-1.3.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `block in require_with_bootsnap_lfi'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/bootsnap-1.3.1/lib/bootsnap/load_path_cache/loaded_features_index.rb:65:in `register'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/bootsnap-1.3.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:20:in `require_with_bootsnap_lfi'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/bootsnap-1.3.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/activesupport-5.2.1/lib/active_support/dependencies/interlock.rb:14:in `block in loading'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/activesupport-5.2.1/lib/active_support/concurrency/share_lock.rb:151:in `exclusive'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/activesupport-5.2.1/lib/active_support/dependencies/interlock.rb:13:in `loading'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/bootsnap-1.3.1/lib/bootsnap/load_path_cache/core_ext/active_support.rb:59:in `depend_on'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/railties-5.2.1/lib/rails/engine.rb:478:in `block (2 levels) in eager_load!'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/railties-5.2.1/lib/rails/engine.rb:477:in `each'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/railties-5.2.1/lib/rails/engine.rb:477:in `block in eager_load!'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/railties-5.2.1/lib/rails/engine.rb:475:in `each'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/railties-5.2.1/lib/rails/engine.rb:475:in `eager_load!'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/railties-5.2.1/lib/rails/engine.rb:356:in `eager_load!'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/railties-5.2.1/lib/rails/application/finisher.rb:69:in `each'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/railties-5.2.1/lib/rails/application/finisher.rb:69:in `block in <module:Finisher>'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/railties-5.2.1/lib/rails/initializable.rb:32:in `instance_exec'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/railties-5.2.1/lib/rails/initializable.rb:32:in `run'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/railties-5.2.1/lib/rails/initializable.rb:61:in `block in run_initializers'
App 16305 output:     /home/ansible_deploy_bot/.rbenv/versions/2.5.1/lib/ruby/2.5.0/tsort.rb:228:in `block in tsort_each'
App 16305 output:     /home/ansible_deploy_bot/.rbenv/versions/2.5.1/lib/ruby/2.5.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
App 16305 output:     /home/ansible_deploy_bot/.rbenv/versions/2.5.1/lib/ruby/2.5.0/tsort.rb:431:in `each_strongly_connected_component_from'
App 16305 output:     /home/ansible_deploy_bot/.rbenv/versions/2.5.1/lib/ruby/2.5.0/tsort.rb:349:in `block in each_strongly_connected_component'
App 16305 output:     /home/ansible_deploy_bot/.rbenv/versions/2.5.1/lib/ruby/2.5.0/tsort.rb:347:in `each'
App 16305 output:     /home/ansible_deploy_bot/.rbenv/versions/2.5.1/lib/ruby/2.5.0/tsort.rb:347:in `call'
App 16305 output:     /home/ansible_deploy_bot/.rbenv/versions/2.5.1/lib/ruby/2.5.0/tsort.rb:347:in `each_strongly_connected_component'
App 16305 output:     /home/ansible_deploy_bot/.rbenv/versions/2.5.1/lib/ruby/2.5.0/tsort.rb:226:in `tsort_each'
App 16305 output:     /home/ansible_deploy_bot/.rbenv/versions/2.5.1/lib/ruby/2.5.0/tsort.rb:205:in `tsort_each'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/railties-5.2.1/lib/rails/initializable.rb:60:in `run_initializers'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/railties-5.2.1/lib/rails/application.rb:361:in `initialize!'
App 16305 output:     /data/websites/psulib_blacklight/config/environment.rb:5:in `<top (required)>'
App 16305 output:     config.ru:3:in `require_relative'
App 16305 output:     config.ru:3:in `block in <main>'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/rack-2.0.5/lib/rack/builder.rb:55:in `instance_eval'
App 16305 output:     /data/websites/psulib_blacklight/vendor/bundle/ruby/2.5.0/gems/rack-2.0.5/lib/rack/builder.rb:55:in `initialize'
App 16305 output:     config.ru:1:in `new'
App 16305 output:     config.ru:1:in `<main>'
App 16305 output:     /usr/share/passenger/helper-scripts/rack-preloader.rb:101:in `eval'
App 16305 output:     /usr/share/passenger/helper-scripts/rack-preloader.rb:101:in `preload_app'
App 16305 output:     /usr/share/passenger/helper-scripts/rack-preloader.rb:189:in `block in <module:App>'
App 16305 output:     /usr/share/ruby/vendor_ruby/phusion_passenger/loader_shared_helpers.rb:380:in `run_block_and_record_step_progress'
App 16305 output:     /usr/share/passenger/helper-scripts/rack-preloader.rb:188:in `<module:App>'
App 16305 output:     /usr/share/passenger/helper-scripts/rack-preloader.rb:30:in `<module:PhusionPassenger>'
App 16305 output:     /usr/share/passenger/helper-scripts/rack-preloader.rb:29:in `<main>'
[ E 2018-09-06 15:11:02.4731 8647/Tv age/Cor/App/Implementation.cpp:221 ]: Could not spawn process for application /data/websites/psulib_blacklight: The application encountered the following error: uninitialized constant Traject (NameError)
  Error ID: b77700d9
  Error details saved to: /tmp/passenger-error-Pe2jP8.html
```

**Comment by banukutlu** [comment]
In GitLab by @cdm32 on Sep 6, 2018, 14:39

changed title from **Uninitialized constant traject** to **Uninitialized constant traject{+ in production+}**

**Comment by banukutlu** [comment]
In GitLab by @cdm32 on Sep 6, 2018, 15:04

changed the description

**Comment by banukutlu** [comment]
In GitLab by @cdm32 on Sep 6, 2018, 15:24

changed the description

**Comment by banukutlu** [comment]
In GitLab by @cdm32 on Sep 7, 2018, 08:24

this one is solved by fixing the downstream bug in `blacklight-marc`

**Comment by banukutlu** [comment]
In GitLab by @cdm32 on Sep 7, 2018, 08:31

added a PR https://github.com/projectblacklight/blacklight-marc/pull/61, for now we'll just use the fork I made, I suppose?

**Comment by banukutlu** [comment]
In GitLab by @cdm32 on Sep 10, 2018, 13:08

added ~938 label

**Comment by banukutlu** [comment]
In GitLab by @cdm32 on Sep 10, 2018, 13:08

created branch [`95-uninitialized-constant-traject-in-production`](https://git.psu.edu/i-tech/psulib_blacklight/compare/master...95-uninitialized-constant-traject-in-production)

**Comment by banukutlu** [comment]
In GitLab by @cdm32 on Sep 10, 2018, 13:13

mentioned in commit f96eebe7af626362fd036720e3780fde764d1762

**Comment by banukutlu** [comment]
In GitLab by @cdm32 on Sep 10, 2018, 13:14

mentioned in merge request !13

**Comment by banukutlu** [comment]
In GitLab by @cdm32 on Sep 10, 2018, 13:15

when https://github.com/projectblacklight/blacklight-marc/pull/61 is merged, we can undo this

**Comment by banukutlu** [comment]
In GitLab by @bzk60 on Sep 10, 2018, 13:23

closed via merge request !13

**Comment by banukutlu** [comment]
In GitLab by @cdm32 on Sep 10, 2018, 13:23

closed via commit f96eebe7af626362fd036720e3780fde764d1762

**Comment by banukutlu** [comment]
In GitLab by @bzk60 on Sep 10, 2018, 13:23

closed via commit 556f999187ca0221da19505e178894d84ce79400

**Comment by banukutlu** [comment]
In GitLab by @bzk60 on Sep 10, 2018, 13:26

removed ~1508 label
